src/array/cpu/spmat_op_impl.cc

```diff
@@ -586,6 +586,59 @@ void CSRSort_(CSRMatrix* csr) {
 template void CSRSort_<kDLCPU, int64_t>(CSRMatrix* csr);
 template void CSRSort_<kDLCPU, int32_t>(CSRMatrix* csr);
 
+template <DLDeviceType XPU, typename IdType, typename TagType>
+IdArray CSRSortByTag_(CSRMatrix* csr, IdArray tag, int64_t num_tags) {
+  auto indptr_data = static_cast<IdType *>(csr->indptr->data);
+  auto indices_data = static_cast<IdType *>(csr->indices->data);
+  auto eid_data = static_cast<IdType *>(csr->data->data);
```

Comment:
It is possible that the csr does not have a data array when the data is simply [0, nnz). You can use `aten::CSRHasData` for check.